### ISSUE TITLE
How does learning rate (lr) change with train process?

### ISSUE THREAD

**Issue by KellyZhao960510** [facebookresearch/wav2letter#658]
![image](https://user-images.githubusercontent.com/30706535/82291947-78fe2d80-99dc-11ea-8b5c-7971225786c4.png)
Above is how my learning rate change with  train process, I found that lr increase with number of  iteration. and Finally it shows that lr always be 0.6, Is 0.6 learning rate too big to converge?

Here is my train command: 
`mpirun --allow-run-as-root -n 4 -mca orte_abort_on_non_zero_status 1 /root/wav2letter/build/Train fork /root/wav2letter/OOCL/saved_models/model.bin --flagsfile train.cfg`

And here is my train flagsfile information:
![image](https://user-images.githubusercontent.com/30706535/82292732-fbd3b800-99dd-11ea-880f-5c976e642ce1.png)

**Comment by tlikhomanenko** [comment]
@KellyZhao960510 

we added the warmup support, which by default 8000 iterations `--warmup`. It is doing linear increase of lr from 0 to value specified in `--lr` (so at the 8000 iteration it will be 0.6) and then your lr will be fixed during time if you do not specify gamma to be less than 1 https://github.com/facebookresearch/wav2letter/blob/master/src/common/Defines.cpp#L104. Other possible things to control lr (decrease at some factor after specific number of iterations) is https://github.com/facebookresearch/wav2letter/blob/master/src/common/Defines.cpp#L89, https://github.com/facebookresearch/wav2letter/blob/master/src/common/Defines.cpp#L93

Warmup is necessary for Transformer archs, if you use something else, you can set warmup to be 0. About lr values - all depends on the optimizer, arch, data and other things. For sgd which you use 0.6 possibly fine, but you need to tune this parameter of course.
